### PR TITLE
[SDKS-7182] Adding telemetry and fetch segments

### PR DESCRIPTION
**Pull request by nmayorsplit**

### CODE REVIEW

File: client/src/main/java/io/split/telemetry/domain/Stats.java [hunk 1]
```diff
@@ -23,6 +23,7 @@ public class Stats {
     /* package private */ static final String FIELD_EVENTS_DROPPED = "eD";
     /* package private */ static final String FIELD_STREAMING_EVENT = "sE";
     /* package private */ static final String FIELD_TAGS = "t";
+    /* package private */ static final String UPDATES_FROM_SSE = "ufs";
```
Comment (gthea) [inline client/src/main/java/io/split/telemetry/domain/Stats.java:26]: nit
```suggestion
    /* package private */ static final String FIELD_UPDATES_FROM_SSE = "ufs";
```

File: client/src/main/java/io/split/telemetry/storage/InMemoryTelemetryStorage.java [hunk 2]
```diff
@@ -209,6 +227,14 @@ public long getSessionLength() {
         return _sdkRecords.get(SdkRecordsEnum.SESSION).get();
     }
 
+    @Override
+    public UpdatesFromSSE popUpdatesFromSSE() {
+        UpdatesFromSSE updatesFromSSE = new UpdatesFromSSE();
+        updatesFromSSE.setSplits(_updatesFromSSERecords.get(UpdatesFromSSEEnum.SPLITS).get());
+        _updatesFromSSERecords.replace(UpdatesFromSSEEnum.SPLITS, new AtomicLong());
```
Comment (gthea) [inline client/src/main/java/io/split/telemetry/storage/InMemoryTelemetryStorage.java:234]: This way the AtomicLong value can be retrieved and updated atomically
```suggestion
        updatesFromSSE.setSplits(_updatesFromSSERecords.get(UpdatesFromSSEEnum.SPLITS).getAndSet(0L));
```

File: client/src/main/java/io/split/telemetry/domain/Stats.java (hunk 1 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/telemetry/domain/Stats.java]: Updated

File: client/src/main/java/io/split/telemetry/storage/InMemoryTelemetryStorage.java (hunk 2 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/telemetry/storage/InMemoryTelemetryStorage.java]: Updated.

**Review by gthea — APPROVED** [review]

**Pull request merged**